File: ctci/1_2.py
one = 'abcda'
two = 'acdaba'
def is_perm(one, two):
    d={char:one.count(char) for char in one}
    for char in two:
        if char not in d.keys() or two.count(char) != d[char]:
            return False
    return True

print(is_perm(one, two))
# ints: #7, #84, #722, #737 

File: ctci/1_2.js
const one = 'abcda';
const two = 'acdab';

const is_perm = (str1, str2) => {
    const strOneObj = [...str1].reduce((res, char) => (res[char] = (res[char] || 0) + 1, res), {})
    for (let property in strOneObj){
        if (!str2.includes(property) || strOneObj[property] != (str2.split(property).length - 1)) {
            return false
        }
    }
    return true
}

console.log(is_perm(one,two))